src/hotspot/cpu/ppc/stubGenerator_ppc.cpp

```diff
@@ -631,6 +631,135 @@ class StubGenerator: public StubCodeGenerator {
     return start;
   }
 
+// Generate stub for ghash process  blocks.
+//
+// Arguments for generated stub:
+//      state:  R3_ARG1
+//      subkeyH:    R4_ARG2
+//      data: R5_ARG3
+//
+address generate_ghash_processBlocks() {
+  StubCodeMark mark(this, "StubRoutines", "ghash");
+  address start = __ function_entry();
+
+  // Registers for parameters
+  Register state = R3_ARG1;                     // long[] state
+  Register subkeyH = R4_ARG2;                   // long[] subH
+  Register data = R5_ARG3;                      // byte[] data
+  Register blocks = R6_ARG4;
+  Register temp1 = R8;
+  Register temp2 = R9;
+  Register temp3 = R10;
+  Register temp4 = R11;
+  Register align = data;
+  // Vector Registers
+  VectorRegister vH = VR0;
+  VectorSRegister vHS = VSR32;
+  VectorRegister vX = VR1;
+  VectorRegister vH_shift = VR2;
+  VectorRegister vTmp1 = VR3;
+  VectorRegister vTmp2 = VR4;
+  VectorRegister vSwappedH = VR5;
+  VectorRegister vTmp4 = VR6;
+  VectorRegister loadOrder = VR7;
+  VectorRegister vMSB = VR8;
+  VectorRegister vLowerH = VR9;
+  VectorRegister vHigherH = VR10;
+  VectorRegister vZero = VR11;
+  VectorRegister vConst1 = VR12;
+  VectorRegister vConst7 = VR13;
+  VectorRegister vConstC2 = VR14;
+  VectorRegister vTmp3 = VR16;
+  VectorRegister vTmp5 = VR17;
+  VectorRegister vTmp6 = VR18;
+  VectorRegister vTmp7 = VR19;
+  VectorRegister vHigh = VR20;
+  VectorRegister vLow = VR21;
+  VectorRegister vPerm = VR22;
+  VectorRegister vZero_Stored = VR23;
+  VectorSRegister vZero_StoredS = VSR55;
+  VectorRegister vMask = VR24;
+  VectorRegister vS = VR25;
+  VectorSRegister vXS = VSR33;
+  Label L_end, L_aligned;
```

Comment:
I suggest to declare `VectorRegisters` only and using `->to_vsr()` below. This should improve readability.

Non-volatile VectorRegisters need to be preserved. See https://github.com/openjdk/jdk/blob/bcb1bdaae772c752d54939dae3a0d95892acc228/src/hotspot/cpu/ppc/macroAssembler_ppc.cpp#L3866